### PR TITLE
Fix broken links in free programming books subjects

### PR DESCRIPTION
**Pull request by ivanvc**
## What does this PR do?
This PR addresses broken links from the free programming subjects page, as reported in #6942.

### Description
N/A

### Why is this valuable (or not)?
N/A

### How do we know it's really free?
N/A

### For book lists, is it a book? For course lists, is it a course? etc.
N/A

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -363,10 +363,10 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Bayesian Reasoning and Machine Learning](http://web4.cs.ucl.ac.uk/staff/D.Barber/pmwiki/pmwiki.php?n=Brml.HomePage)
 * [Deep Learning](http://www.deeplearningbook.org) - Ian Goodfellow, Yoshua Bengio, Aaron Courville
 * [Deep Learning for Coders with Fastai and PyTorch](https://github.com/fastai/fastbook) - Jeremy Howard, Sylvain Gugger (Jupyter Notebooks)
-* [Deep Learning with PyTorch](https://pytorch.org/assets/deep-learning/Deep-Learning-with-PyTorch.pdf) - Eli Stevens, Luca Antiga, Thomas Viehmann (PDF)
+* [Deep Learning with PyTorch](https://livebook.manning.com/book/deep-learning-with-pytorch/) - Eli Stevens, Luca Antiga, Thomas Viehmann
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:366]: only free for a minute or so. just remove it

Comment (ivanvc) [reply books/free-programming-books-subjects.md:366]: @eshellman I saw it's available at The Internet Archive when the site was cached. Should we keep that copy? Or should I remove it?

**Comment by ivanvc** [comment]
@eshellman I saw that the Deep Learning with PyTorch eBook is available at The Internet Archive when the site was cached. Should we keep that copy? Or should I remove it?

File: books/free-programming-books-subjects.md (hunk 1 above)
Comment (eshellman) [reply books/free-programming-books-subjects.md:366]: remove it

**Pull request merged**